rir/src/interpreter/interp.cpp

```diff
@@ -3052,11 +3124,15 @@ SEXP evalRirCode(Code* c, InterpreterInstance* ctx, SEXP* env,
 
 eval_done:
     if (synthesizeFrames) {
+        SEXP e = *env;
         while (!synthesizeFrames->empty()) {
             FrameInfo* f = synthesizeFrames->front();
             synthesizeFrames->pop_front();
             SEXP res = ostack_pop(ctx);
-            SEXP e = ostack_pop(ctx);
+            // Inlined functions may or may not have contexts
+            if (e == R_GlobalContext->cloenv)
```

Comment:
I do not clearly see why this condition couldn't hold while we did not create an context?

Reply:
superseeded by changes